components/compose/awesomebar/src/main/java/mozilla/components/compose/browser/awesomebar/internal/Suggestions.kt

```diff
@@ -35,14 +36,29 @@ internal fun Suggestions(
         state = state,
         modifier = Modifier.testTag("mozac.awesomebar.suggestions")
     ) {
-        items(suggestions) { suggestion ->
-            Suggestion(
-                suggestion,
-                colors,
-                orientation,
-                onSuggestionClicked,
-                onAutoComplete
-            )
+        groups.forEach { group ->
+            val groupSuggestions = suggestions[group] ?: emptyList()
+
+            if (groupSuggestions.isNotEmpty()) {
+                group.title?.let { title ->
+                    item(group.id) {
+                        SuggestionGroup(title, colors)
+                    }
+                }
+
+                items(
+                    items = groupSuggestions,
+                    key = { suggestion -> "${group.id}:${suggestion.provider.id}:${suggestion.id}" }
```

Comment:
what's the key here used for?

Reply:
It's an optimization. When the list of items changes then this allows the compose runtime to identify the rows and know whether some of them have just moved or whether they are new rows. See:
https://developer.android.com/jetpack/compose/lists#item-keys